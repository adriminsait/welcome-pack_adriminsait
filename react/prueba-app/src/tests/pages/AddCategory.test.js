import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'

import { AddCategory } from '../../pages/AddCategory';

describe('pruebas <AddCategory>', () => {
  
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    })

    test('probando que se ve', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('probando cambios en input', () => {

        const input = wrapper.find('input');
        const value = 'hola mundo'

        input.simulate('change', {
            target: {value}
        });

        // expect(wrapper).toMatchSnapshot();
    });

    test('No debe hacer nada el submit', () => {

        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        })

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('debe llamar a setCategories', () => {

        const value = 'hola mundo'

        wrapper.find('input').simulate('change', {
            target: {value}
        });
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        })

        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');
    });
    
    
});
