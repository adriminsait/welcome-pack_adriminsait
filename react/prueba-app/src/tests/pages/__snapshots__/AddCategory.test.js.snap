// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas <AddCategory> probando que se ve 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
